tiledb/api/src/error.rs

```diff
@@ -185,6 +199,15 @@ pub enum Error {
     Other(String),
 }
 
+impl Error {
+    pub(crate) fn physical_type_mismatch<T, U>() -> Self {
```

Comment:
#[cfg(test)]

Reply:
It is also used for the proptest-strategies feature, i.e. `cargo check --features proptest-strategies` fails to compile without it.

We can do `#[cfg(any(test, feature = "proptest-strategies"))]` as we do in other places, but I do anticipate that we will find a use for it somewhere that does not require this configuration.